### PR TITLE
Always use localhost, not host name

### LINKED ISSUE #2350: influxd -hostname localhost
`influxd -hostname localhost` stopped working after #2233 was merged

https://github.com/influxdb/influxdb/pull/2333/files#r28724886

### PR DESCRIPTION
**Pull request by corylanou**
Fixes #2350

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,8 @@
 - [#2338](https://github.com/influxdb/influxdb/pull/2338): Fix panic if tag key isn't double quoted when it should have been
 - [#2340](https://github.com/influxdb/influxdb/pull/2340): Fix SHOW DIAGNOSTICS panic if any shard was non-local.
 - [#2351](https://github.com/influxdb/influxdb/pull/2351): Fix data race by rlocking shard during diagnostics.
+- [#2350](https://github.com/influxdb/influxdb/pull/2350): Issue fix for :influxd -hostname localhost.
```
Comment (otoolep) [inline CHANGELOG.md]: @corylanou -- this is not the usual style we've been following, unless you are also proposing to merge 2350?

Comment (otoolep) [inline CHANGELOG.md]: #2350 is an issue, not a PR. The usual style is not to add issue URL, but just the PR URL, and say something like "fixes 2350". Unless I am missing something?

Comment (corylanou) [inline CHANGELOG.md]: I was told that based on what Jason/Todd talked about last week, we now add each issue that it fixes, along with the PR.

Comment (otoolep) [inline CHANGELOG.md]: OK. I thought Jason's change was a once-off, and the format was more like:

#1234 <URL> description.....fixes 4678, 3789, 23789

where 1234 is a PR, not an issue.

I think the link is wrong though in your change. It works, but it's actually an issue.

**Comment by otoolep** [comment]
+1 on green build.

**Comment by neonstalwart** [comment]
what about https://github.com/influxdb/influxdb/blob/30b56cea0383f7c72b5be5574a08da63750b9dc8/cmd/influxd/config.go#L257-L259 also?

File: cmd/influxd/config.go [hunk 2]
```diff
@@ -253,11 +252,6 @@ func NewConfig() *Config {
 	c.Broker.MaxTopicSize = DefaultBrokerMaxTopicSize
 	c.Broker.MaxSegmentSize = DefaultBrokerMaxSegmentSize
 
-	// Detect hostname (or set to localhost).
-	if c.Hostname, _ = os.Hostname(); c.Hostname == "" {
-		c.Hostname = "localhost"
-	}
-
```
Comment (neonstalwart) [inline cmd/influxd/config.go]: wondering why it wouldn't be good to still have

```go
if c.Hostname == "" {
  c.Hostname = "localhost"
}
```

i'm probably not thinking about something.

Comment (corylanou) [inline cmd/influxd/config.go]: `c.Hostname` is always `""` when you create a new config.

**Comment by toddboom** [comment]
+1

**Pull request merged**